src/components/form/select.js

```diff
@@ -27,570 +24,476 @@ import { helpers } from '../../common';
+    if (convertedOption === undefined || convertedOption === null) {
       convertedOption = {
-        title: option,
+        title: `${option}`,
         value: option
       };
+    }
 
-      updatedOptions[index] = convertedOption;
-    } else if (typeof convertedOption.title === 'function') {
-      convertedOption.title = convertedOption.title();
+    if (_isPlainObject(convertedOption)) {
+      convertedOption.title = convertedOption.title || 'undefined';
+
+      switch (convertedOption.value) {
+        case undefined:
+        case null:
+          break;
+        case '':
+          convertedOption.value = '';
+      }
+
+      convertedOption.isSelected = convertedOption.isSelected || false;
+      convertedOption.index = index;
+      convertedOption.key = `${helpers.generateHash({ value: convertedOption.value, title: convertedOption.title })}`;
     }
 
-    convertedOption.text = convertedOption.text || convertedOption.title;
-    convertedOption.textContent = convertedOption.textContent || convertedOption.title;
-    convertedOption.label = convertedOption.label || convertedOption.title;
+    return convertedOption;
+  });
+};
+
+/**
+ * A memoized response for the updateOptions function. Assigned to a property for testing function.
+ */
+updateOptions.memo = helpers.memo(updateOptions);
+
+/**
+ * Update selected list options for consumption.
+ *
+ * @param {string|number|null|NaN|{value: unknown}|
+ *     Array<string|number|null|undefined|NaN|{value:
+ *     unknown}>} options
+ * @returns {Array|Array<"NaN"|"null"|unknown>}
+ */
+const updateSelectedOptions = options =>
+  (Array.isArray(options) && options) ||
+  (options !== undefined && options !== null && !Number.isNaN(options) && [options]) ||
+  (options === null && ['null']) ||
+  (Number.isNaN(options) && ['NaN']) ||
+  [];
+
+/**
+ * A memoized response for the updateSelectedOptions function. Assigned to a property for testing function.
+ */
+updateSelectedOptions.memo = helpers.memo(updateSelectedOptions, { cacheLimit: 25 });
 
-    if (activateOptions.length) {
-      let isSelected;
+/**
+ * Update the isSelected property for formatted options.
+ *
+ * @param {object} params
+ * @param {Array<{title, value, isSelected}>} params.options
+ * @param {Array} params.selectedOptions
+ * @param {SelectVariant|string} params.variant
+ * @returns {{isSelected:boolean}|undefined|Array<{isSelected:boolean}>|Array}
+ */
+const updateSelectedProp = ({ options, selectedOptions = [], variant = SelectVariant.single } = {}) =>
+  options
+    .map(option => {
+      const { isSelected, title, value } = option;
+      let updateIsSelected = isSelected;
+
+      if (updateIsSelected === true) {
+        return option;
+      }
 
-      if (_isPlainObject(convertedOption.value)) {
-        isSelected = _findIndex(activateOptions, convertedOption.value) > -1;
+      if (_isPlainObject(value)) {
```

Comment:
required: looks like our checks are skipping this, expand the test to check for an `object` in value, ie `{ value: { data: 'some data' } }`